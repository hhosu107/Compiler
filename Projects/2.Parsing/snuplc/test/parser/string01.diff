2,3c2,32
< got 8:12: tString (Hello, world\n)
< parse error : at 8:12 : factor expected.
---
> successfully parsed.
>   AST:
>     CAstScope: 'string01'
>       symbol table:
>         [[
>           [ *DIM(<ptr(4) to <NULL>>,<int>) --> <int>           ]
>           [ *DOFS(<ptr(4) to <NULL>>) --> <int>           ]
>           [ *ReadInt() --> <int>           ]
>           [ *WriteChar(<char>) --> <NULL>           ]
>           [ *WriteInt(<int>) --> <NULL>           ]
>           [ *WriteLn() --> <NULL>           ]
>           [ *WriteStr(<ptr(4) to <array  of <char>>>) --> <NULL>           ]
>           [ @_str_1   <array 14  of <char>>           ]
>             [ data: 'Hello, world\n' ]
>           [ @_str_2   <array 23  of <char>>           ]
>             [ data: 'This is pretty nice!\n\n' ]
>         ]]
>       statement list:
>         call [ *WriteStr(<ptr(4) to <array  of <char>>>) --> <NULL> ] <NULL>
>           &() <ptr(4) to <array 14  of <char>>>
>             "Hello, world\n" <array 14  of <char>>
>         call [ *WriteStr(<ptr(4) to <array  of <char>>>) --> <NULL> ] <NULL>
>           &() <ptr(4) to <array 23  of <char>>>
>             "This is pretty nice!\n\n" <array 23  of <char>>
>       nested scopes:
>         empty.
>     
> 
> 
> run the following command to convert the .dot file into a PDF:
>   dot -Tpdf -ostring01.mod.ast.pdf string01.mod.ast.dot
