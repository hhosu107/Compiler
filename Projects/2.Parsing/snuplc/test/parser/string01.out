parsing 'string01.mod'...
got 8:12: tString (Hello, world\n)
parse error : at 8:12 : factor expected.


Done.
